devito/interfaces.py

```diff
@@ -1,29 +1,102 @@
 
     def __new__(cls, *args, **kwargs):
         return IndexedBase.__new__(cls, args[0], shape=args[1])
 
+    # this is the default directory where memmap files will be created
+    _default_disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+    # default exit code used on interrupt and kill
+    _default_exit_code = 0
+
+    # functions for managing memmap files
+    # call this method to specify where you want memmap file to be created
+    @staticmethod
+    def set_default_disk_path(disk_path):
+        """call this method to specify default directory to create memmap file"""
+        if not os.path.exists(disk_path):
+            os.makedirs(disk_path)
+        DenseData._default_disk_path = disk_path
+        print("default disk path set to: " + DenseData._default_disk_path)
+
+    @staticmethod
+    def get_memmap_file_list():
+        """returns the list of string path to all memmap file created"""
+        return DenseData._memmap_file_list
+
+    # used for clean memmap file on normal exit
+    @staticmethod
+    def remove_memmap_file():
+        """call this method at the end of script to clean up generated memmpa file"""
+        for f in DenseData._memmap_file_list:
+            try:
+                os.remove(f)
+            except OSError:
+                print("error removing " + f + " skipping")
+                pass
+
+    # used for clean memmap file on signal, internal method
+    @staticmethod
+    def _remove_memmap_file_on_signal(*args):
+        DenseData.remove_memmap_file()
+        sys.exit(DenseData._default_exit_code)
+
+    # to register clean up method for chosen signals
+    @staticmethod
+    def register_remove_memmap_file_signal():
+        """call this method to make sure memmap files are removed on interrupt or kill"""
+        for sig in (SIGABRT, SIGINT, SIGSEGV, SIGTERM):
+            signal(sig, DenseData._remove_memmap_file_on_signal)
+    # end of functions for managing memmap files
+
+    # This function allocate memmory for this data. if either _defualt_disk_path or
+    # self.disk_path is not None, a numpy memmap is used, if not a numpy ndarray is used.
+    # Warning: this function assume that seld.name is unique
+    def _allocate_memory(self):
+        """allocate memmory for this data. if either _defualt_disk_path or self.disk_path is not None,
+           a numpy memmap is used, if not a numpy ndarray is used."""
+        if DenseData._default_disk_path is None and self.disk_path is None:
+            # not disk_path use ndarray
+            self.pointer = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+            return
+        elif self.disk_path is None:
+            # using defualt disk_path
+            self.disk_path = DenseData._default_disk_path
+        elif not os.path.exists(self.disk_path):
+            # create disk path
+            os.makedirs(self.disk_path)
```

Comment:
Sure. The thing is that creation of object specific path must be directly or indirectly associate with the object's class. Since we cannot predict what object gets created at runtime.